apis/v1beta1/object_reference_types.go

```diff
@@ -99,9 +99,22 @@ type BackendObjectReference struct {
 	// +kubebuilder:default=""
 	Group *Group `json:"group,omitempty"`
 
-	// Kind is kind of the referent. For example "HTTPRoute" or "Service".
+	// Kind is the Kubernetes resource kind of the referent. For example
+	// "Service" or "HTTPRoute".
+	//
 	// Defaults to "Service" when not specified.
 	//
+	// ExternalName services can refer to CNAME DNS records that may live
+	// outside of the cluster and as such are difficult to reason about in
+	// terms of conformance. Implementations are recommended to not support
+	// ExternalName Services when feasible. As such we consider ExternalName
+	// Services to be supported at an "Implementation-specific" level for usage
+	// as a BackendObjectReference.
```

Comment:
Can we start with a stronger statement here? If someone really wants to support ExternalName services we can revisit this at that point.
```suggestion
	// terms of conformance. They also may not be safe to forward to (see 
	// CVE-2021-25740 for more information). Implementations SHOULD NOT
	// support ExternalName Services.
```

Reply:
When we state "Implementations SHOULD NOT support ExternalName Service." I guess we need tests to assert such behavior in our conformance suite. But if we really mean to block this possibility, we should assert that all the ways to mimic `externalName` (as with `endpointslices` with FQDN entries) are blocked and declared as not conformant as well.

Reply:
> Can we start with a stronger statement here? If someone really wants to support ExternalName services we can revisit this at that point.

Sounds good, changed in bef0c2087ba2c5e69b69a5689a9c7c1799b32b86 :+1: 

> When we state "Implementations SHOULD NOT support ExternalName Service." I guess we need tests to assert such 
> behavior in our conformance suite. But if we really mean to block this possibility, we should assert that all the ways to
> mimic externalName (as with endpointslices with FQDN entries) are blocked and declared as not conformant as well.

MUST NOT is something that we would definitely want to add conformance assertions for, but not SHOULD NOT. SHOULD NOT is saying "here be dragons" and giving a clear sign to implementers that they travel to these dragon infested waters at their own risk, yarrrr

Reply:
Gotcha, it makes sense! I think that if we'd write documentation in this way, instead of the usual boring one, more people would read it :)